src/apps/filters/h3.c

```diff
@@ -1632,6 +1632,811 @@ SUBCOMMAND(cellsToMultiPolygon,
+    // for it
+    bool isValid = H3_EXPORT(isValidDirectedEdge)(cell);
+    if (!isValid) {
+        return E_DIR_EDGE_INVALID;
+    }
+    double length = 0;
+    H3Error err = H3_EXPORT(edgeLengthRads)(cell, &length);
+    if (err) {
+        return err;
+    }
+    printf("%.10lf\n", length);
+    return E_SUCCESS;
+}
+
+SUBCOMMAND(edgeLengthKm,
+           "The exact edge length of a specific directed edge in kilometers") {
+    DEFINE_CELL_ARG(cell, cellArg);
+    Arg *args[] = {&edgeLengthKmArg, &cellArg, &helpArg};
+    PARSE_SUBCOMMAND(argc, argv, args);
+    // This one is pretty loose about the inputs it accepts, so let's validate
+    // for it
+    bool isValid = H3_EXPORT(isValidDirectedEdge)(cell);
+    if (!isValid) {
+        return E_DIR_EDGE_INVALID;
+    }
+    double length = 0;
+    H3Error err = H3_EXPORT(edgeLengthKm)(cell, &length);
+    if (err) {
+        return err;
+    }
+    printf("%.10lf\n", length);
+    return E_SUCCESS;
+}
+
+SUBCOMMAND(edgeLengthM,
+           "The exact edge length of a specific directed edge in meters") {
+    DEFINE_CELL_ARG(cell, cellArg);
+    Arg *args[] = {&edgeLengthMArg, &cellArg, &helpArg};
+    PARSE_SUBCOMMAND(argc, argv, args);
+    // This one is pretty loose about the inputs it accepts, so let's validate
+    // for it
+    bool isValid = H3_EXPORT(isValidDirectedEdge)(cell);
+    if (!isValid) {
+        return E_DIR_EDGE_INVALID;
+    }
+    double length = 0;
+    H3Error err = H3_EXPORT(edgeLengthM)(cell, &length);
+    if (err) {
+        return err;
+    }
+    printf("%.10lf\n", length);
+    return E_SUCCESS;
+}
+
+SUBCOMMAND(getNumCells,
+           "The number of unique H3 cells for a specified resolution") {
+    int res = 0;
+    Arg resArg = {.names = {"-r", "--resolution"},
+                  .required = true,
+                  .scanFormat = "%d",
+                  .valueName = "res",
+                  .value = &res,
+                  .helpText = "Resolution, 0-15 inclusive."};
+    Arg *args[] = {&getNumCellsArg, &resArg, &helpArg};
+    PARSE_SUBCOMMAND(argc, argv, args);
+    int64_t numCells = 0;
+    H3Error err = H3_EXPORT(getNumCells)(res, &numCells);
+    if (err) {
+        return err;
+    }
+    printf("%" PRId64 "\n", numCells);
+    return E_SUCCESS;
+}
+
+SUBCOMMAND(getRes0Cells, "Returns all of the resolution 0 cells") {
+    Arg *args[] = {&getRes0CellsArg, &helpArg};
+    PARSE_SUBCOMMAND(argc, argv, args);
+    H3Index *out = calloc(122, sizeof(H3Index));
+    H3Error err = H3_EXPORT(getRes0Cells)(out);
+    if (err != E_SUCCESS) {
+        free(out);
+        return err;
+    }
+    printf("[");
+    bool hasPrinted = false;
+    for (int i = 0; i < 122; i++) {
+        if (out[i] > 0) {
+            if (hasPrinted) {
+                printf(", ");
+            }
+            printf("%" PRIx64, out[i]);
+            hasPrinted = true;
+        }
+    }
+    printf("]\n");
```

Comment:
For `polygonToCells` we print newline-delimited cells, but most of the multi-cell outputs here use JSON arrays. Should this be consistent?